src/pydap/lib.py

```diff
@@ -288,7 +309,7 @@ def fix_shorthand(projection, dataset):
         if len(var) == 1 and var[0][0] not in list(dataset.keys()):
             token, slice_ = var.pop(0)
             for child in walk(dataset):
-                if token == child.name:
+                if token == unquote(child.name):
```

Comment:
I found this necessary. Otherwise when there was a projection of a variable name, say "foo[", child.name was escaped but "foo[" in `token` (taken form the query string) was not being recognized to be part of the original dataset. An alternative was to escape (or quote) token in the comparison. That also worked. The variable name in `token` is later escaped.